api/edge_api/identities/views.py

```diff
@@ -300,6 +306,29 @@ def all(self, request, *args, **kwargs):
 
         return Response(serializer.data)
 
+    @swagger_auto_schema(request_body=EdgeIdentitySourceIdentityRequestSerializer())
+    @action(detail=False, methods=["POST"], url_path="clone-from-given-identity")
+    def clone_from_given_identity(self, request, *args, **kwargs) -> Response:
+        """
+        Clone feature states from a given source identity.
+        """
+        # Get and validate source identity
+        serializer = EdgeIdentitySourceIdentityRequestSerializer(
+            data=request.data, context={"request": request}
+        )
+        serializer.is_valid(raise_exception=True)
+
+        source_identity: EdgeIdentity = self.get_identity(
+            edge_identity_identity_uuid=request.data.get("source_identity_uuid")
+        )
```

Comment:
Nit: Use the serializer to get the `source_identity_uuid` since it's been validated.